code/modules/economy/quikpay.dm

```diff
@@ -0,0 +1,245 @@
+	var/obj/item/weapon/paper/R = new(src.loc)
+	R.name = "Quik And Easy: How to make a transaction"
+
+	R.info += "<b>Quik-Pay setup:</b><br>"
+	R.info += "<ol><li>Remember your access code included on the paper that is included with your device</li>"
+	R.info += "<li>Enter the pin into the locking mechanism to be able to add items to the menu</li>"
+	R.info += "<li>Add items to the menu by typing the item name and its price</li></ol>"
+	R.info += "<b>When starting a new transaction:</b><br>"
+	R.info += "<ol><li>Have the customer enter the amount of the item they want.</li>"
+	R.info += "<li>Press confirm payment</li>"
+	R.info += "<li>Allow them to review the sum.</li>"
+	R.info += "<li>Have them swipe their card to pay for the items, press return to reset the sum.</li></ol>"
+
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.offset_x += 0
+	R.offset_y += 0
+	R.ico += "paper_stamp-cent"
+	R.stamped += /obj/item/weapon/stamp
+	R.add_overlay(stampoverlay)
+	R.stamps += "<HR><i>This paper has been stamped by the Head of Personnel's desk.</i>"
+
+/obj/item/device/nanoquikpay/AltClick(var/mob/user)
+	var/obj/item/weapon/card/id/I = user.GetIdCard()
+	if(istype(I) && (access_heads in I.access))
+		editmode = 1
+		to_chat(user, span("notice", "Command access granted."))
+		SSvueui.check_uis_for_change(src)
+
+
+/obj/item/device/nanoquikpay/proc/print_reference()
+	var/obj/item/weapon/paper/R = new(src.loc)
+	var/pname = "Reference: [machine_id]"
+	var/info = "<b>[machine_id] reference</b><br><br>"
+	info += "Access code: [access_code]<br><br>"
+	info += "<b>Do not lose or misplace this code.</b><br>"
+	R.set_content_unsafe(pname, info)
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.stamped += /obj/item/weapon/stamp
+	R.add_overlay(stampoverlay)
+	R.stamps += "<HR><i>This paper has been stamped by the Head of Personnel's desk.</i>"
+	var/obj/item/smallDelivery/D = new(R.loc)
+	R.forceMove(D)
+	D.wrapped = R
+	D.name = "small parcel - 'Quik Pay access code'"
+
+
+/obj/item/device/nanoquikpay/proc/print_receipt()
+	var/obj/item/weapon/paper/R = new(usr.loc)
+	var/receiptname = "Receipt: [machine_id]"
+	R.set_content_unsafe(receiptname, receipt, sum)
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.stamped += /obj/item/weapon/stamp
+	R.add_overlay(stampoverlay)
+	R.stamps += "<HR><i>This paper has been stamped by the Quik-Pay device.</i>"
+
+
+
+/obj/item/device/nanoquikpay/attackby(obj/O, mob/user)
+	var/obj/item/weapon/card/id/I = O.GetID()
+	if (!I) 
+		return
+	if (!istype(O))
+		return
+
+	var/datum/money_account/customer_account = SSeconomy.get_account(I.associated_account_number)
```

Comment:
The money transfer should use:
`/datum/controller/subsystem/economy/proc/transfer_money(var/source_number, var/dest_number, var/purpose, var/terminal_id, var/amount, var/source_pin=null, var/mob/user=null)`